### PR TITLE
Close #27795: Add metrics ping to see if user allows notifications

### LINKED ISSUE #27795: Add metrics ping to see if user allows notifications
For Android 13, notification is not allowed by default,  adding telemetry to find out how many users have notification disabled with Fenix.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21969)

### PR DESCRIPTION
**Pull request by rocketsroger**
Tested with emulators on API 33, API 29 and API 25 to ensure that when notification/marketing channel is enabled/disabled, the probe reported values matching our expectations.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #27795

### CODE REVIEW

**Comment by rocketsroger** [comment]
# Request for data collection review form

All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.

1. What questions will you answer with this data?
  * This data will help us know how many users allow notifications from us.

2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? 
  * This will allow us to know more about user engagement with notifications.
 
3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
  * There are no other alternatives. 
4. Can current instrumentation answer these questions?
  * No. 
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki. 
Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
Notifications allowed:  If user allow notifications from us.  True if notifications are not blocked, otherwise false | Category 1 - technical data | https://github.com/mozilla-mobile/fenix/issues/27795


6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
  * https://dictionary.telemetry.mozilla.org/apps/fenix 
7. How long will this data be collected? Choose one of the following:
  * Up until version 122, with the option to renew at that point.

8. What populations will you measure?
  * All channels, all locales, all countries 
9. If this data collection is default on, what is the opt-out mechanism for users?
  * Default Glean SDK opt-out mechanism. 
10. Please provide a general description of how you will analyze this data.
  * Glean and Glam. 
11. Where do you intend to share the results of your analysis?
  * Only on Glean, Glam, and with mobile teams. 
12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
   * No third-party tools.

**Comment by eliserichards** [comment]
> # Request for data collection review form
> 
> All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.
> 
>     1. What questions will you answer with this data?
> 
> 
>     * This data will help us know how many users allow notifications from us.
> 
> 
>     2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
> 
> 
>     * This will allow us to know more about user engagement with notifications.
> 
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
> 
> 
>     * There are no other alternatives.
> 
> 
>     4. Can current instrumentation answer these questions?
> 
> 
>     * No.
> 
> 
>     5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.    
>        Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.
> 
> 
> Measurement Description 	Data Collection Category 	Tracking Bug #
> Notifications allowed: If user allow notifications from us. True if notifications are not blocked, otherwise false 	Category 1 - technical data 	#27795
> 
>     6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
> 
> 
>     * [https://dictionary.telemetry.mozilla.org/apps/fenix   ](https://dictionary.telemetry.mozilla.org/apps/fenix%E2%80%A8)
> 
> 
>     7. How long will this data be collected? Choose one of the following:
> 
> 
>     * Up until version 122, with the option to renew at that point.
> 
> 
>     8. What populations will you measure?
> 
> 
>     * All channels, all locales, all countries
> 
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
>     * Default Glean SDK opt-out mechanism.
> 
> 
>     10. Please provide a general description of how you will analyze this data.
> 
> 
>     * Glean and Glam.
> 
> 
>     11. Where do you intend to share the results of your analysis?
> 
> 
>     * Only on Glean, Glam, and with mobile teams.
> 
> 
>     12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
> 
> 
>     * No third-party tools.


# Data Review Form (to be filled by Data Stewards)

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

    * Yes, through the metrics.yaml and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/)

2) Is there a control mechanism that allows the user to turn the data collection on and off? (Note, for data collection not needed for security purposes, Mozilla provides such a control mechanism) Provide details as to the control mechanism available.

    * Yes, through the "Send Usage Data" preference in the application settings.

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

    * No, the data will be collected until major release version 122 with the option to remove/renew at that time.

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

    * Category 1 - technical data

5) Is the data collection request for default-on or default-off?

    * Default-on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

    * No

7) Is the data collection covered by the existing Firefox privacy notice? 

    * Yes

8) Does the data collection use a third-party collection tool? **If yes, escalate to legal.**

    * No

## Result

data-review+

File: app/metrics.yaml [hunk 1]
```diff
@@ -1752,6 +1752,24 @@ metrics:
     metadata:
       tags:
         - Wallpapers
+  notifications_allowed:
+    type: boolean
+    description: |
+      True if notifications are not blocked, otherwise false.
```
Comment (csadilek) [inline app/metrics.yaml:1758]: nit: are allowed. (Easier to read).

File: app/src/main/java/org/mozilla/fenix/FenixApplication.kt [hunk 2]
```diff
@@ -714,6 +715,15 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
...
 
             defaultWallpaper.set(isDefaultTheCurrentWallpaper)
+
+            @Suppress("TooGenericExceptionCaught")
+            try {
+                notificationsAllowed.set(
+                    NotificationManagerCompat.from(applicationContext).areNotificationsEnabled(),
+                )
+            } catch (e: Exception) {
+                Logger.warn("Failed to check if notification is enabled", e)
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/FenixApplication.kt:725]: nit: if notifications are enabled.

**Review by csadilek — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/FenixApplication.kt [hunk 3]
```diff
@@ -714,6 +715,15 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
                 Wallpaper.nameIsDefault(settings.currentWallpaperName)
 
             defaultWallpaper.set(isDefaultTheCurrentWallpaper)
+
+            @Suppress("TooGenericExceptionCaught")
+            try {
+                notificationsAllowed.set(
+                    NotificationManagerCompat.from(applicationContext).areNotificationsEnabled(),
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/FenixApplication.kt:722]: This is good to have, but I've realized that we should also record if we the specific channel we're interested in is enabled (and possibly do that directly in the worker). There's a different API for that via the notification channel. Let's add that on Monday before landing. Happy to help with that.

Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/FenixApplication.kt:722]: Sounds good.  I will look into doing that. Thanks

**Review by csadilek — APPROVED** [review]

**Comment by rocketsroger** [comment]
> # Request for data collection review form
> 
> All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.
> 
>     1. What questions will you answer with this data?
> 
> 
>     * This data will help us know how many users allow notifications from us.
> 
> 
>     2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
> 
> 
>     * This will allow us to know more about user engagement with notifications.
> 
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
> 
> 
>     * There are no other alternatives.
> 
> 
>     4. Can current instrumentation answer these questions?
> 
> 
>     * No.
> 
> 
>     5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.    
>        Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.
> 
> 
> Measurement Description 	Data Collection Category 	Tracking Bug #
> Marketing channel enabled: Is marketing notification channel enabled. True if marketing channel is enabled, otherwise false 	Category 1 - technical data 	#27795
> Notifications allowed: If user allow notifications from us. True if notifications are not blocked, otherwise false 	Category 1 - technical data 	#27795
> 
>     6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
> 
> 
>     * [https://dictionary.telemetry.mozilla.org/apps/fenix   ](https://dictionary.telemetry.mozilla.org/apps/fenix%E2%80%A8)
> 
> 
>     7. How long will this data be collected? Choose one of the following:
> 
> 
>     * Up until version 122, with the option to renew at that point.
> 
> 
>     8. What populations will you measure?
> 
> 
>     * All channels, all locales, all countries
> 
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
>     * Default Glean SDK opt-out mechanism.
> 
> 
>     10. Please provide a general description of how you will analyze this data.
> 
> 
>     * Glean and Glam.
> 
> 
>     11. Where do you intend to share the results of your analysis?
> 
> 
>     * Only on Glean, Glam, and with mobile teams.
> 
> 
>     12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
> 
> 
>     * No third-party tools.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/apps/fenix )

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, through the "Send Usage Data" preference in the application settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

N/A, collection set to end or be renewed by version 122

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 1, Technical data

5) Is the data collection request for default-on or default-off?

default-on

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+

**Comment by rocketsroger** [comment]
@Mergifyio backport releases_v108.0.0

**Pull request merged**